app/src/main/java/net/bible/service/readingplan/ReadingPlanDao.kt

```diff
@@ -196,43 +212,80 @@ class ReadingPlanDao {
             try {
                 // check to see if a user has created his own reading plan with this name
                 val userReadingPlanFile = File(USER_READING_PLAN_FOLDER, filename)
                 val isUserPlan = userReadingPlanFile.exists()
 
                 if (!isUserPlan) {
-                    inputStream = assetManager.open(READING_PLAN_FOLDER + File.separator + filename)
+                    inputStreamRaw = assetManager.open(READING_PLAN_FOLDER + File.separator + filename)
                 } else {
-                    inputStream = FileInputStream(userReadingPlanFile)
+                    inputStreamRaw = FileInputStream(userReadingPlanFile)
                 }
-                properties.load(inputStream)
+
+                val byteArrayForReuse = ByteArrayOutputStream().apply {write(inputStreamRaw?.readBytes())}
+                properties.load(ByteArrayInputStream(byteArrayForReuse.toByteArray()))
+                properties.planCode = planCode
+                properties.numberOfPlanDays = getNumberOfPlanDays(properties)
+                properties.versification = getReadingPlanVersification(properties)
+                getNameAndDescFromProperties(ByteArrayInputStream(byteArrayForReuse.toByteArray()), properties)
+
                 Log.d(TAG, "The properties are now loaded")
                 Log.d(TAG, "properties: $properties")
 
                 // cache it so we don't constantly reload the properties
-                cachedPlanCode = planCode
                 cachedPlanProperties = properties
 
             } catch (e: IOException) {
                 System.err.println("Failed to open reading plan property file")
                 e.printStackTrace()
             } finally {
-                IOUtil.close(inputStream)
+                IOUtil.close(inputStreamRaw)
             }
         }
         return cachedPlanProperties!!
     }
 
+    private fun getNameAndDescFromProperties(inputStream: InputStream, properties: ReadingPlanProperties) {
+        var lineCount = 0
+        var loopCount = 0
+        // Get first commented lines from file for Plan Name (first line)
+        // and Description (following commented lines) up to line 5: otherwise any
+        // commented lines further down in the file will also get added to description
+        inputStream.bufferedReader().forEachLine {
+            if (it.startsWith("#") && loopCount < 5) {
+                val lineWithoutCommentMarks: String = it.trim().replaceFirst("^(\\s*#*\\s*)".toRegex(), "")
+                Log.d(TAG, lineWithoutCommentMarks)
+                if (lineCount == 0) {
+                    properties.planName = StringUtils.trim(lineWithoutCommentMarks)
+                } else {
+                    properties.planDescription = StringUtils.trim(
+                        "${properties.planDescription ?: ""} $lineWithoutCommentMarks "
+                    )
+                }
+                lineCount++
+            }
+            loopCount++
```

Comment:
if loopCount >= 5, we could break and return?

Reply:
I thought so too but seems there is no break that works for "forEachLine" method.

https://stackoverflow.com/a/48623119/877453

That answer says return@forEachLine works, but I tried all kinds of stuff (including this I think) when I wrote this and could not get them working at all.